### PR TITLE
[ASM][IAST] Reflection Injection

### PR DESCRIPTION
**Pull request by e-n-0**
## Summary of changes
Created aspects for `System.Activator` and `System.Type` and run a detection of tainted strings that are passed as parameters to these methods and check the object array for methods that take arguments.

## Reason for change
Detection of Reflection injection vulnerability.
https://datadoghq.atlassian.net/wiki/spaces/APS/pages/3435364372/Reflection+injection

## Test coverage
- Tests of all methods of: (that are available for their frameworks version)
    - Activator: `System.Activator::CreateInstance`, `System.Activator::CreateInstanceFrom` and `System.Activator::CreateComInstanceFrom`
    - Type: `System.Type::GetType`, `System.Type::InvokeMember`

- Using `TestMethodsAspectCover` and `TestAllAspectsHaveACorrespondingMethod` to test that all aspects are used

### CODE REVIEW

File: tracer/src/Datadog.Trace/Configuration/IntegrationId.cs [hunk 1]
```diff
@@ -73,6 +73,7 @@ internal enum IntegrationId
         UnvalidatedRedirect,
         TestPlatformAssemblyResolver,
         StackTraceLeak,
-        XpathInjection
+        XpathInjection,
+        SystemReflection,
```
Comment (daniel-romano-DD) [inline tracer/src/Datadog.Trace/Configuration/IntegrationId.cs:77]: IMHO ReflectionInjection should be more correct, as done with the rest of the vuln integrations

**Review by daniel-romano-DD — APPROVED** [review]
LGTM with a minor note. Thanks a lot

File: tracer/src/Datadog.Trace/Configuration/IntegrationId.cs (hunk 1 above)
Comment (e-n-0) [reply tracer/src/Datadog.Trace/Configuration/IntegrationId.cs]: fixed b5b0003

**Comment by andrewlock** [comment]
## Execution-Time Benchmarks Report :stopwatch:

Execution-time results for samples comparing the following branches/commits:
- [This PR (5219)](https://github.com/DataDog/dd-trace-dotnet/tree/b5b00033dd8d82309b1bacf5f6a6a758d784150e)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/a707b89fb16088d183246e9b2e20031bfb709269)

Execution-time benchmarks measure the whole time it takes to execute a program. And are intended to measure the one-off costs. Cases where the execution time results for the PR are worse than latest master results are shown in **red**. The following thresholds were used for comparing the execution times:
* Welch test with statistical test for significance of **5%**
* Only results indicating a difference greater than **5%** and **5 ms** are considered.


Note that these results are based on a _single_ point-in-time result for each branch. For full results, see the [dashboard](https://ddstaging.datadoghq.com/dashboard/4qn-6fi-54p/apm-net-execution-time-benchmarks).

Graphs show the p99 interval based on the mean and StdDev of the test run, as well as the mean value of the run (shown as a diamond below the graph).
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5219) - mean (74ms)  : 65, 83
     .   : milestone, 74,
    master - mean (71ms)  : 66, 77
     .   : milestone, 71,

    section CallTarget+Inlining+NGEN
    This PR (5219) - mean (977ms)  : 961, 994
     .   : milestone, 977,
    master - mean (975ms)  : 951, 999
     .   : milestone, 975,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5219) - mean (110ms)  : 106, 114
     .   : milestone, 110,
    master - mean (111ms)  : 107, 115
     .   : milestone, 111,

    section CallTarget+Inlining+NGEN
    This PR (5219) - mean (717ms)  : 692, 742
     .   : milestone, 717,
    master - mean (715ms)  : 689, 741
     .   : milestone, 715,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5219) - mean (94ms)  : 90, 98
     .   : milestone, 94,
    master - mean (93ms)  : 90, 97
     .   : milestone, 93,

    section CallTarget+Inlining+NGEN
    This PR (5219) - mean (671ms)  : 645, 698
     .   : milestone, 671,
    master - mean (663ms)  : 641, 686
     .   : milestone, 663,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5219) - mean (187ms)  : 181, 192
     .   : milestone, 187,
    master - mean (187ms)  : 184, 191
     .   : milestone, 187,

    section CallTarget+Inlining+NGEN
    This PR (5219) - mean (1,053ms)  : 1029, 1078
     .   : milestone, 1053,
    master - mean (1,058ms)  : 1040, 1076
     .   : milestone, 1058,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5219) - mean (270ms)  : 263, 277
     .   : milestone, 270,
    master - mean (270ms)  : 264, 275
     .   : milestone, 270,

    section CallTarget+Inlining+NGEN
    This PR (5219) - mean (1,053ms)  : 1025, 1081
     .   : milestone, 1053,
    master - mean (1,054ms)  : 1033, 1075
     .   : milestone, 1054,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5219) - mean (258ms)  : 254, 263
     .   : milestone, 258,
    master - mean (259ms)  : 252, 266
     .   : milestone, 259,

    section CallTarget+Inlining+NGEN
    This PR (5219) - mean (994ms)  : 973, 1016
     .   : milestone, 994,
    master - mean (992ms)  : 973, 1012
     .   : milestone, 992,

```

**Pull request merged**